src/main/java/fr/techad/edc/httpd/search/ContentSearcher.java

```diff
@@ -89,12 +93,37 @@ public List<DocumentationSearchResult> search(String search, String lang, int li
       results.add(documentationSearchResult);
 
     }
-    if (results.isEmpty() && !defaultLanguage.equals(lang)) {
+    if (results.isEmpty() && !defaultLanguage.equals(lang) && checkLanguageIsPresent(lang) == false) {
       return search(search, defaultLanguage, limit, exact, defaultLanguage);
     }
     return results;
   }
 
+  private boolean checkLanguageIsPresent(String lang) throws IOException {
+    Optional<File> product = getProduct(this.config);
+    String parsed;
+    if (product.isPresent()) {
+      parsed = FileUtils.readFileToString(new File(product.get().getCanonicalPath() + "/info.json"),
+              StandardCharsets.UTF_8.name());
+      JSONObject obj = new JSONObject(parsed);
+      String languages = obj.get("languages").toString();
+      List<String> nameList = new LinkedList<String>();
+      nameList.add(obj.get("languages").toString());
+
+      return languages.contains(lang);
+    }
+    return false;
+  }
+
+  public static Optional<File> getProduct(WebServerConfig config) {
```

Comment:
This method is misnamed. It does not return a product but a pointer to the language definition file.
It should not be defined in this class and the static should be limited. Its use shows a design error.